Source/WebCore/editing/cocoa/AttributedString.mm

```diff
@@ -85,79 +89,202 @@
+}
 
-        auto ensureTableResult = tables.ensure(tableID, [&] {
-            return tableBlock.table;
+inline static RetainPtr<NSParagraphStyle> reconstructStyle(const ParagraphStyle& style, IdentifierToTableMap& tables, IdentifierToTableBlockMap& tableBlocks, IdentifierToListMap& lists)
+{
+    for (const auto& item : style.textLists) {
+        lists.ensure(item.thisID, [&] {
+            RetainPtr list = adoptNS([[PlatformNSTextList alloc] initWithMarkerFormat:item.markerFormat options:0]);
+            [list setStartingItemNumber:item.startingItemNumber];
+            return list;
         });
+    }
 
-        auto table = ensureTableResult.iterator->value;
-        RetainPtr<NSTextTableBlock> replacementBlock;
-        if (!ensureTableBlockResult.isNewEntry)
-            replacementBlock = tableBlock;
-        else if (!ensureTableResult.isNewEntry) {
-            replacementBlock = adoptNS([[PlatformNSTextTableBlock alloc] initWithTable:table startingRow:tableBlock.startingRow rowSpan:tableBlock.rowSpan startingColumn:tableBlock.startingColumn columnSpan:tableBlock.columnSpan]);
-            [replacementBlock _takeValuesFromTextBlock:tableBlock];
-            tableBlocks.set(tableBlockID, replacementBlock.get());
-        }
+    for (const auto& item : style.textTables) {
+        tables.ensure(item.thisID, [&] {
+            RetainPtr table = adoptNS([(NSTextTable *)[PlatformNSTextTable alloc] init]);
+            [table setNumberOfColumns:item.numberOfColumns];
+
+            NSTextTableLayoutAlgorithm layout;
+            switch (item.layout) {
+            case TextTableLayoutAlgorithm::Automatic:
+                layout = NSTextTableAutomaticLayoutAlgorithm;
+                break;
+            case TextTableLayoutAlgorithm::Fixed:
+                layout = NSTextTableFixedLayoutAlgorithm;
+                break;
+            }
+
+            [table setLayoutAlgorithm:layout];
+            [table setCollapsesBorders:item.collapsesBorders];
+            [table setHidesEmptyCells:item.hidesEmptyCells];
+
+            const ParagraphStyleCommonTableAttributes& attributes = item;
+            configureNSTextBlockFromParagraphStyleCommonTableAttributes(table.get(), attributes);
+
+            return table;
+        });
+    };
+
+    for (const auto& item : style.textTableBlocks) {
+        tableBlocks.ensure(item.thisID, [&] {
+            auto foundTable = tables.find(item.tableID);
+            if (foundTable == tables.end()) {
+                RELEASE_LOG_ERROR(Editing, "Table not found when trying to reconstruct NSParagraphStyle");
+                return adoptNS([[PlatformNSTextTableBlock alloc] init]);
+            }
+
+            RetainPtr tableBlock = adoptNS([[PlatformNSTextTableBlock alloc] initWithTable:foundTable->value.get() startingRow:item.startingRow rowSpan:item.rowSpan startingColumn:item.startingColumn columnSpan:item.columnSpan]);
+
+            NSTextBlockVerticalAlignment verticalAlignment;
```

Comment:
Ditto, initialize this to a default value for if item.verticalAlignment is out of bounds.  Same in a few other places.